scripts/post-comment.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# Copied from ember-data https://github.com/emberjs/data/blob/master/scripts/asset-size-tracking/src/post-comment.sh
+# Shamelessly stolen from the very awesome ShakingFingerAction
+#  Written by the amazing https://twitter.com/jessfraz
+#  https://github.com/jessfraz/shaking-finger-action
+# Generified to enable this form of comment management
+#  for our CI Reports (AssetSize | Perf Analysis etc.)
+
+if [[ -z "$COMMENT_MARKER" ]]; then
+  echo "Set the COMMENT_MARKER env variable."
+  return 1
+fi
+
+if [[ -z "$COMMENT_TEXT" ]]; then
+  echo "Set the COMMENT_TEXT env variable."
+  return 1
+fi
+
+if [[ -z "$GITHUB_REPOSITORY" ]]; then
+  echo "Set the GITHUB_REPOSITORY env variable."
+  return 1
+fi
+
+# Github seems to be playin coy, getting access from
+# a sourced script is extremely hard
+if [[ -z "${GITHUB_TOKEN}" ]]; then
+  echo "Configure the env to include the GITHUB_TOKEN secret variable."
+  return 1
+fi
+
+URI=https://api.github.com
+API_VERSION=v3
+API_HEADER="Accept: application/vnd.github.${API_VERSION}+json; application/vnd.github.antiope-preview+json"
+AUTH_HEADER="Authorization: token ${GITHUB_TOKEN}"
+
+FOUND_EXISTING=1
+update_comment_if_exists() {
+  # Get all the comments for the pull request.
+  body=$(curl -sSL -H "${AUTH_HEADER}" -H "${API_HEADER}" "${URI}/repos/${GITHUB_REPOSITORY}/issues/${NUMBER}/comments")
```

Comment:
Future task: I'd like to see this script converted to using `gh` (which is setup and authorized) instead of `curl`.

Reply:
:+1: thats a great thing for someone in data to do as well, I'll open an issue